server/routes/incidentDetails/incidentDetailsPage.ts

```diff
@@ -0,0 +1,728 @@
+      incitedInput = currentAssociatedPrisonerNumber
+    }
+  }
+  const data = {
+    incidentDate: getIncidentDate(pageData.formData.incidentDetails?.incidentDate),
+    locationId: pageData.formData.incidentDetails?.locationId,
+    originalRadioSelection: radioSelectionCodeFromIncidentRole(currentIncidentRoleSelection),
+    associatedPrisonersNumber: pageData.formData.incidentDetails?.currentAssociatedPrisonerNumber,
+    associatedPrisonersName: pageData.displayData.associatedPrisonersName,
+  }
+  return res.render(`pages/incidentDetails`, {
+    errors: error ? [error] : [],
+    prisoner: pageData.displayData.prisoner,
+    locations: pageData.displayData.possibleLocations,
+    data,
+    reportingOfficer: pageData.displayData.reporter || '',
+    submitButtonText: 'Save and continue',
+    exitButtonHref,
+    incitedInput,
+    assistedInput,
+    originalIncidentRoleSelection: pageData.formData.originalIncidentRoleSelection,
+    lastIncidentRoleSelection: pageData.formData.lastIncidentRoleSelection,
+    lastAssociatedPrisonerNumberSelection: pageData.formData.lastAssociatedPrisonerNumberSelection,
+    originalReporterUsername: pageData.formData.originalReporterUsername,
+  })
+}
+
+export const updateDataOnSearchReturn = (
+  stashedData: StashedIncidentDetails,
+  requestData: RequestValues
+): StashedIncidentDetails => {
+  if (!requestData.selectedPerson) {
+    return stashedData
+  }
+  return {
+    incidentDetails: {
+      ...stashedData.incidentDetails,
+      currentAssociatedPrisonerNumber: requestData.selectedPerson,
+    },
+    temporaryData: stashedData.temporaryData,
+  }
+}
+
+export const updateDataOnDeleteReturn = (
+  stashedData: StashedIncidentDetails,
+  requestData: RequestValues
+): StashedIncidentDetails => {
+  if (!requestData.deleteWanted || requestData.deleteWanted !== 'true') {
+    return stashedData
+  }
+  return {
+    incidentDetails: {
+      currentAssociatedPrisonerNumber: null,
+      ...stashedData.incidentDetails,
+    },
+    temporaryData: stashedData.temporaryData,
+  }
+}
+
+const getDraftIdFromString = (draftId: string): number => {
+  let draftIdValue: number = null
+  if (draftId) {
+    draftIdValue = parseInt(draftId, 10)
+  }
+  return draftIdValue
+}
+
+const radioSelectionCodeFromIncidentRole = (incidentRole: IncidentRole): string => {
+  if (!incidentRole) {
+    return null
+  }
+  let radioSelectionCode = null
+  switch (incidentRole) {
+    case IncidentRole.COMMITTED:
+      radioSelectionCode = 'committed'
+      break
+    case IncidentRole.ATTEMPTED:
+      radioSelectionCode = 'attempted'
+      break
+    case IncidentRole.INCITED:
+      radioSelectionCode = 'incited'
+      break
+    case IncidentRole.ASSISTED:
+    // Fall through
+    default:
+      radioSelectionCode = 'assisted'
+      break
+  }
+  return radioSelectionCode
+}
+
+const incidentRoleFromRadioSelection = (radioSelectionCode: string): IncidentRole => {
```

Comment:
I think this *might* be possible without the if statement - the IncidentRole enum has these values encoded in them - it feels like it should be able to reverse. I might have a play